containers/ome-tiff-offsets/context/main.py

```diff
@@ -2,26 +2,35 @@
 from glob import glob
 from pathlib import Path
 from os import makedirs
+from os.path import splitext
 import json
+from os.path import dirname
+from os.path import exists
+from os import makedirs
 
 from skimage.external import tifffile
 
-
 def get_offsets(tiff_filepath):
     with tifffile.TiffFile(tiff_filepath) as tif:
         offsets = [page._offset for page in tif.pages]
     return offsets
 
 def main(input_dir, output_dir):
     makedirs(output_dir, exist_ok=True)
-    for input_path in glob(input_dir + '/*.ome.tiff') + glob(input_dir + '/*.ome.tif'):
+
+    # Find all OME.TIFFs in the input directory.
+    for input_path in sum([Path(input_dir).glob('**/*.ome.{ext}') for ext in ['tif', 'tiff'], []]):
         offsets = get_offsets(input_path)
-        # Truncate `.ome.tiff`.
-        input_name = Path(input_path).with_suffix('').with_suffix('').name
-        output_path = Path(output_dir) / str(input_name + '.offsets.json')
+
+        # Create output path for each OME.TIFF:
+        output_path = dirname(str(Path(output_dir) / Path(input_path).relative_to(input_dir)))
```

Comment:
Converting `input_dir` and `output_dir` to `pathlib.Path` here seems a little clumsy, especially doing it more than once. It's probably better to adjust the calls to `parser.add_argument` to add `type=Path`.

Reply:
@mruffalo I agree the code can be improved. At this point I am trying to build a pipeline that works which later can be cleaned/generalized. Do you recommend doing it now?

Reply:
@mruffalo I took advantage of your suggestion since it was a quick fix.